pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
+	if !validations.FileExists(directory) {
+		if err := os.Mkdir(directory, dirPermission); err != nil {
+			return fmt.Errorf("unable to create directory %s", directory)
+		}
+	}
+
+	for _, pack := range packages {
+		displayPackage := NewDisplayPackage(pack)
+		content, err := yaml.Marshal(displayPackage)
+		if err != nil {
+			fmt.Println(fmt.Errorf("unable to parse package %s %v", pack.Name, err).Error())
+			continue
+		}
+		err = writeToFile(directory, pack.Name, content)
+		if err != nil {
+			fmt.Println(err.Error())
```

Comment:
```suggestion
			fmt.Fprintln(os.Stderr, err.Error())
			continue // or return an error to abort.
```
This is just good hygiene in case someone adds code below later.